bundles/org.openhab.core.config.xml/src/main/java/org/openhab/core/config/xml/AbstractXmlBasedProvider.java

```diff
@@ -173,9 +177,12 @@ private T_OBJECT acquireLocalizedObject(Bundle bundle, T_OBJECT object, @Nullabl
             return cacheEntry;
         }
 
+        @Nullable
         final T_OBJECT localizedObject = localize(bundle, object, locale);
         if (localizedObject != null) {
-            localizedObjectCache.put(localizedKey, localizedObject);
+            @NonNull
+            T_OBJECT nonNullLocalizedObject = (@NonNull T_OBJECT) localizedObject;
```

Comment:
The compiler failed without explicitly casting `localizedObject`. If it's not assigned here and casted as parameter on the line below, the cast would be removed by the auto save actions, making the compiler error again. Using `@format:on/off` didn't help. :slightly_frowning_face: